core/src/main/java/org/opensearch/sql/calcite/CalciteRelNodeVisitor.java

```diff
@@ -2375,6 +2394,152 @@ private String getValueFunctionName(UnresolvedExpression aggregateFunction) {
     return sb.toString();
   }
 
+  @Override
+  public RelNode visitChart(Chart node, CalcitePlanContext context) {
+    visitChildren(node, context);
+    ArgumentMap argMap = ArgumentMap.of(node.getArguments());
+    List<UnresolvedExpression> groupExprList =
+        Stream.of(node.getRowSplit(), node.getColumnSplit()).filter(Objects::nonNull).toList();
+    ChartConfig config = ChartConfig.fromArguments(argMap);
+    Aggregation aggregation =
+        new Aggregation(
+            List.of(node.getAggregationFunction()), List.of(), groupExprList, null, List.of());
+    BitSet nonNullGroupMask = new BitSet(groupExprList.size());
+    // Rows without a row-split are always ignored
+    if (config.useNull) {
+      nonNullGroupMask.set(0);
+    } else {
+      nonNullGroupMask.set(0, groupExprList.size());
+    }
+    visitAggregation(aggregation, context, false, nonNullGroupMask);
+    RelBuilder relBuilder = context.relBuilder;
+
+    // If row or column split does not present or limit equals 0, this is the same as `stats agg
+    // [group by col]` because all truncating is performed on the column split
```

Comment:
row or column split does not present or limit equals 0, chart and stats produce different results, 

```
[
    {"_time":1755622404, "age":"10", "id":"1", "deptno":"a"},
    {"_time":1755622405, "age":10, "id":"2", "deptno":"a"},
    {"_time":1755622408, "age":"str", "id":"1", "deptno":"b"},
    {"_time":1755622409, "age":"str", "id":"2", "deptno":"b"}
]
```
* chart should ignore null value
```
| chart limit=0 sum(age) by id, deptno

id, deptno, sum
1, a, 10
2, a, 10
```

* stats keep null value
```
| stats sum(age) by id, depto

id, deptno, sum
1, a, 10
1, b, null
2, a, 10
2, b, null
```